python_modules/dagster/dagster/_cli/api.py

```diff
@@ -328,7 +326,10 @@ def execute_step_command(input_json, compressed_input_json):
         args = deserialize_value(input_json, ExecuteStepArgs)
 
         with get_instance_for_cli(instance_ref=args.instance_ref) as instance:
-            dagster_run = instance.get_run_by_id(args.run_id)
+            dagster_run = check.not_none(
+                instance.get_run_by_id(args.run_id),
+                f"Run with id '{args.run_id}' not found for step execution",
+            )
```

Comment:
this is just moving the non-null check from below to a better spot (preventing any unexpected nulls from propagating)